### PR TITLE
tweaks to `Stack.alloc_words`

### PR DESCRIPTION
**Pull request by ggreif**
This incorporates some review feedback from #3793.

### CODE REVIEW

**Comment by crusso** [comment]
Clever, but actually, is this code every even generated? It's only if you statically allocate more than a page and I doubt we ever do that, and almost certainly not often?

**Comment by ggreif** [comment]
When done without the page size test, I get
- rw-r--r-- 1 ggreif staff 458276 Feb 16 14:56 /Users/ggreif/motoko/test/run/_out/aardvark.wasm
- rw-r--r-- 1 ggreif staff 458272 Feb 16 15:14 /Users/ggreif/motoko/test/run/_out/aardvark.wasm (this PR)
So there would be a slight improvement :-)

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -1172,27 +1172,28 @@ module Stack = struct
       (* read last word of reserved page to force trap *)
       compile_unboxed_const 0xFFFF_FFFCl ^^
       G.i (Load {ty = I32Type; align = 2; offset = 0l; sz = None}) ^^
-      G.i Drop
+      G.i Unreachable
```
[the remark this answers is not in the document]
Comment (crusso) [reply src/codegen/compile.ml:1175]: I like that.

**Comment by crusso** [comment]
But I'm less enamored with the overflow check, since its less clear and doesn't seem to buy us anything beyond obfuscation on a rare path.

File: src/codegen/compile.ml [hunk 2]
```diff
@@ -1172,27 +1172,28 @@ module Stack = struct
...
     G.i (Binary (Wasm.Values.I32 I32Op.Sub)) ^^
     set_stack_ptr env ^^
-    get_stack_ptr env
+    get_stack_ptr env ^^
+    (* check for stack overflow, if necessary *)
+    if n_bytes >= page_size then
+      get_stack_ptr env ^^
+      G.i (Unary (Wasm.Values.I32 I32Op.Clz)) ^^
+      G.if0
+        G.nop
```
Comment (ggreif) [inline src/codegen/compile.ml:1193]:
```suggestion
        G.nop (* we found leading zeros, i.e. no wraparound *)
```

**Review by crusso — APPROVED** [review]
Somewhat reluctantly

**Pull request merged**